Arcane Engine Core/src/input/JoystickInputData.h

```diff
@@ -0,0 +1,78 @@
+/*
+	Author: Mohamed Kazma
+	Descriptions: Holds joystick data and handles events sent from the joystick manager via glfw
+	and updates the corresponding data it is holding
+*/
+
+#pragma once
+
+#define MAX_JOYSTICK_BUTTONS 18
+
+#define ARCANE_GAMEPAD_A 0
+#define ARCANE_GAMEPAD_B 1
+#define ARCANE_GAMEPAD_X 2
+#define ARCANE_GAMEPAD_Y 3
+#define ARCANE_GAMEPAD_LEFT_BUMPER 4
+#define ARCANE_GAMEPAD_RIGHT_BUMPER 5
+#define ARCANE_GAMEPAD_BACK 6
+#define ARCANE_GAMEPAD_START 7
+#define ARCANE_GAMEPAD_GUIDE 8
+#define ARCANE_GAMEPAD_LEFT_THUMB 9
+#define ARCANE_GAMEPAD_RIGHT_THUMB 10
+#define ARCANE_GAMEPAD_DPAD_UP 11
+#define ARCANE_GAMEPAD_DPAD_RIGHT 12
+#define ARCANE_GAMEPAD_DPAD_DOWN 13
+#define ARCANE_GAMEPAD_DPAD_LEFT 14
+#define ARCANE_GAMEPAD_LAST   ARCANE_GAMEPAD_DPAD_LEFT
+#define ARCANE_GAMEPAD_CROSS ARCANE_GAMEPAD_A
+#define ARCANE_GAMEPAD_CIRCLE   ARCANE_GAMEPAD_B
+#define ARCANE_GAMEPAD_SQUARE   ARCANE_GAMEPAD_X
+#define ARCANE_GAMEPAD_TRIANGLE   ARCANE_GAMEPAD_Y
+
+namespace arcane
+{
+	class JoystickInputData
+	{
+	public:
+		// Store the state of each joytick button to be queried
+		unsigned char m_ButtonStates[MAX_JOYSTICK_BUTTONS];
+
+		JoystickInputData(char id = 0);
+		~JoystickInputData();
+
+		// Update the joystick's connection status
```

Comment:
The comments for each of these functions are quite redundant. I would personally get rid of them. Since you named the functions well.

Reply:
dunnzoooo